pandas/tests/frame/conftest.py

```diff
@@ -0,0 +1,121 @@
+import pytest
```

Comment:
A couple of things about this file:

* Do you use all of these fixtures in your changes?
* I would prefer if the naming is a little more consistent e.g.:
- You have the word "frame" is some of your fixture names but not othres
- You have underscores between words in some names but not others

Reply:
> This is based on #22236, and so it will be easier to review once that is in.

I'm not sure I follow you here.  How does adding currently unused fixtures in this PR get explained in #22236?  I'm sensing you might want to break some of the fixtures for a separate PR.

Reply:
It's buried in the review there and related to #22471 - replacing the attributes of `TestData` with fixtures. I did it all in one go as a start towards #22471. It's a direct translation of the attributes, including their names.

Reply:
Hmm...I'm sorry, but I think this is a little too much for one PR.  Keep in mind that the purpose of the PR, per your title, is better error reporting for `df.set_index`.  This is a lot of new code that is arguably not pertinent to that aim.

Reply:
The diff of this PR will be tiny after #22236. Feel free to comment there about the changes of that PR

Reply:
FYI, I literally have the same questions in that PR as I do in this one for that file.

Reply:
I'll answer in more detail in the other thread. But I already mentioned what it's about:

> It's [...] related to #22471 - replacing the attributes of `TestData` with fixtures. I did it all in one go as a start towards #22471. It's a direct translation of the attributes, including their names.